runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -639,6 +660,118 @@ else if (event.getMessage().contains(KILLED_TELEBLOCK_OPPONENT_TEXT))
 				}
 			}
 		}
+
+		if (config.showTzhaarTimers())
+		{
+			String message = Text.removeTags(event.getMessage());
+			Matcher matcher = TZHAAR_COMPLETE_MESSAGE.matcher(message);
+
+			if (message.contains(TZHAAR_DEFEATED_MESSAGE) || matcher.matches())
+			{
+				removeTzhaarTimer();
+				resetTzhaarConfig();
+				resetTzhaarVars();
+				return;
+			}
+
+			Instant now = Instant.now();
+			matcher = TZHAAR_PAUSED_MESSAGE.matcher(message);
```

Comment:
Rather than reassigning `matcher` multiple times, using distinct names for each matcher would be much clearer when reading this branch.

Reply:
I believe the original version of my code used distinct names and Kamiel recommended using just one matcher variable in early review of the plugin when cleaning up the code, so I'd like to confirm this is actually what is wanted before i change it, again.